### PR TITLE
Fix README

### LINKED ISSUE #1030: README.md example is broken
I wanted to update the README.md example to match the current "reality". Unfortunately, it seems to have been broken by the modules changes by @AlexD97 and @jankoboehm (CC @fieker). I am now getting this:
```
julia> k, a = quadratic_field(-5)
(Imaginary quadratic field defined by x^2 + 5, sqrt(-5))

julia> zk = maximal_order(k)
Maximal order of Imaginary quadratic field defined by x^2 + 5
with basis nf_elem[1, sqrt(-5)]

julia> factorisations(zk(6))
2-element Vector{Fac{NfOrdElem}}:
 -1 * (sqrt(-5) + 1) * (sqrt(-5) - 1)
 -1 * 2 * -3

julia> Qx, x = PolynomialRing(QQ, [:x1,:x2])
(Multivariate Polynomial Ring in x1, x2 over Rational Field, fmpq_mpoly[x1, x2])

julia> R = grade(Qx, [1,2])[1]
Multivariate Polynomial Ring in x1, x2 over Rational Field graded by
  x1 -> [1]
  x2 -> [2]

julia> f = R(x[1]^2+x[2])
x1^2 + x2

julia> degree(f)
graded by [2]

julia> F = FreeModule(R, 1)
Free module of rank 1 over R

julia> s = sub(F, [f*F[1]])
(Submodule over R with 1 generator and no relations
, Module homomorphism with
Domain: Submodule over R with 1 generator and no relations

Codomain: F)

julia> H, mH = hom(s, quo(F, s))
ERROR: MethodError: no method matching quo(::AbstractAlgebra.Generic.FreeModule{MPolyElem_dec{fmpq, fmpq_mpoly}}, ::Tuple{AbstractAlgebra.Generic.Submodule{MPolyElem_dec{fmpq, fmpq_mpoly}}, AbstractAlgebra.Generic.ModuleHomomorphism{MPolyElem_dec{fmpq, fmpq_mpoly}}})
Closest candidates are:
  quo(::S, ::Hecke.AbsAlgAssIdl{S, T, U}) where {S, T, U} at ~/.julia/packages/Hecke/M7nVW/src/AlgAss/Ideal.jl:428
  quo(::AbstractAlgebra.FPModule{T}, ::AbstractAlgebra.FPModule{T}) where T<:RingElement at ~/.julia/packages/AbstractAlgebra/KPT2D/src/QuotientModule.jl:22
Stacktrace:
 [1] top-level scope
   @ REPL[11]:1
```

Perhaps PR #1021 helps ? I didn't try yet.

### PR DESCRIPTION
**Pull request by thofma**

### CODE REVIEW

**Comment by fingolfin** [comment]
There are more changes, though, and at least one further error downwards:

```
julia> D = grading_group(H)
ERROR: MethodError: no method matching grading_group(::SubQuo{MPolyElem_dec{fmpq, fmpq_mpoly}})
Closest candidates are:
  grading_group(::MPolyRing_dec) at ~/Projekte/OSCAR/Oscar.jl/src/Rings/mpoly-graded.jl:49
  grading_group(::MPolyQuo{<:MPolyElem_dec}) at ~/Projekte/OSCAR/Oscar.jl/src/Rings/MPolyQuo.jl:939
Stacktrace:
 [1] top-level scope
   @ REPL[22]:1
```

I'll update this PR a bit, but don't know how to resolve that error

**Comment by fingolfin** [comment]
Perhaps there should be a definition like `grading_group(q::SubQuo) = grading_group(base_ring(q))` ??? Dunno if that makes sense... CC @jankoboehm @AlexD97 @fieker @wdecker

**Comment by AlexD97** [comment]
SubQuo is not graded, so it makes no sense to call grading_group() on a SubQuo.

File: README.md [hunk 1]
```diff
@@ -105,14 +95,10 @@ Domain:
...
+1 -> e[1]
+by Submodule with 1 generator
 1 -> (x1^2 + x2)*e[1]
-defined on the Singular side
-
-
-
 
 julia> D = grading_group(H)
 GrpAb: Z
```
Comment (thofma) [inline README.md]:
```suggestion
```

**Comment by fingolfin** [comment]
@AlexD97 but this used to work, right? So I hope what you really mean is that subquotients no longer support gradings but these will be added back later?

**Comment by thofma** [comment]
I deleted it. If someone cares about it, he can add it back.

**Comment by AlexD97** [comment]
> @AlexD97 but this used to work, right? So I hope what you really mean is that subquotients no longer support gradings but these will be added back later?

Yes, exactly. In future, SubQuo_dec will be back.

**Pull request merged**